changefeedccl: webhook sink concurrency implementation and memory monitoring

spiffyy99 opened the pull request:
re-merging previously merged PR's #67028 #67369, were reverted because of test flakes which were fixed by #67499

Epic CRDB-7748

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/67539)
<!-- Reviewable:end -->

spiffyy99 committed:
changefeedccl: implement concurrent execution for webhook sink

Previously, messages sent to the webhook sink were sent synchronously
one after another. This change improves performance by using concurrent
worker goroutines to send sink messages while maintaining ordering
guarantees.

Resolves #67009

Release note: None

stevendanna commented on pkg/ccl/changefeedccl/sink_webhook.go:
```diff
@@ -187,9 +201,40 @@ func makeWebhookClient(u sinkURL, timeout time.Duration) (*httputil.Client, erro
+		case <-s.ctx.Done():
+			return
+		case msg := <-workerCh:
+			err := s.sendMessage(s.ctx, msg)
```
Do we want to add a TODO here to document the ordering issue that we identified as needing to be addressed?

spiffyyeng replied:
done!

stevendanna commented on pkg/ccl/changefeedccl/nemeses_test.go:
```diff
@@ -47,7 +47,7 @@ func TestChangefeedNemeses(t *testing.T) {
 	t.Run(`enterprise`, enterpriseTest(testFn, feedTestNoTenants))
 	t.Run(`cloudstorage`, cloudStorageTest(testFn, feedTestNoTenants))
-	t.Run(`http`, webhookTest(testFn, feedTestNoTenants))
+	t.Run(`webhook`, webhookTest(testFn, feedTestNoTenants))
```
I believe you mentioned this being potentially flaky, was that resolved?

spiffyyeng replied:
Yes, flakes like this (timeouts) are happening all over builds because of an http2 issue: https://github.com/cockroachdb/cockroach/issues/67658

stevendanna reviewed (approved):
Approving, but I left a couple of questions

spiffyy99 committed:
changefeedccl: add memory monitoring for webhook sink inflight messages

Resolves #67309

Release note: None

spiffyyeng commented:
bors r+
